x/dsmr/node_test.go

```diff
@@ -961,11 +1000,57 @@ func TestNode_BuildBlock_IncludesChunks(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	r.NoError(node.BuildChunk(
+		context.Background(),
+		[]dsmrtest.Tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	))
+
+	blk, err := node.BuildBlock(context.Background(), node.LastAccepted, 2)
+	r.NoError(err)
+	r.NoError(node.Verify(context.Background(), node.LastAccepted, blk))
+	_, err = node.Accept(context.Background(), blk)
+	r.NoError(err)
+
+	_, err = node.BuildBlock(context.Background(), node.LastAccepted, 3)
+	r.ErrorIs(err, ErrNoAvailableChunkCerts)
+
+	// make sure that it's not the case with any other chunk.
+	r.NoError(node.BuildChunk(
+		context.Background(),
+		[]dsmrtest.Tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	))
+	_, err = node.Accept(context.Background(), blk)
```

Comment:
We should never call `Accept` on the same block twice

Reply:
good point. This Accept was not needed, so I've just dropped it.